libs/bits/bit_array.go

```diff
@@ -145,12 +144,9 @@ func (bA *BitArray) And(o *BitArray) *BitArray {
 	if bA == nil || o == nil {
 		return nil
 	}
+	o = o.Copy()
```

Comment:
Maybe this should be AFTER bA.mtx.RLock, to ensure that neither bA nor o changed during execution of this function

Reply:
answered above